dd-java-agent/agent-bootstrap/src/main/java/datadog/trace/bootstrap/instrumentation/java/concurrent/ConcurrentState.java

```
@@ -32,7 +32,7 @@ private ConcurrentState() {}
   public static <K> ConcurrentState captureScope(
       ContextStore<K, ConcurrentState> contextStore, K key, AgentScope scope) {
     if (scope != null && scope.isAsyncPropagating()) {
-      if (scope.span() instanceof AgentTracer.NoopAgentSpan) {
+      if (scope.span() == noopSpan()) {
```

Are we sure we don't want blackhole spans to also stop propagation here?

---

Today is what happens being a subclass of `NoopAgentSpan`. If we let them propagate I think the major risk is that the user will believe that the tracing is restored by closing a blackhole scope while this can activate in other threads. I would have perhaps kept todays's behaviour and don't propagate